src/components/graph/graph.coords.js

```diff
@@ -0,0 +1,333 @@
+import CONST from "./graph.const";
+import { logWarning } from "../../utils";
+
+/**
+ * Computes the normalized vector from a vector.
+ * @param {Object} vector a 2D vector with x and y components
+ * @param {number} vector.x x coordinate
+ * @param {number} vector.y y coordinate
+ * @returns {Object} normalized vector
+ * @memberof Graph/helper
+ */
+function normalize(vector) {
+  const norm = Math.sqrt(Math.pow(vector.x, 2) + Math.pow(vector.y, 2));
+  return norm === 0 ? vector : { x: vector.x / norm, y: vector.y / norm };
+}
+
+/**
+ * Calculate the length of a vector, from the center of a rectangle,
+ * to one of it's edges.
+ * This calculation is taken from https://stackoverflow.com/a/3197924.
+ *
+ * @param {Object.<string, number>} RectangleCoords, The coords of the left-top vertex, and the right-bottom vertex.
+ * @param {Object.<string, number>} VectorOriginCoords The center of the rectangle coords.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components.
+ * @returns {number} The length of the vector from the center of the symbol to it's closet edge, considering the given direction vector.
+ */
+function calcRectangleVectorLengthFromCoords({ x1, y1, x2, y2 }, { px, py }, directionVector) {
+  const angle = Math.atan(directionVector.y / directionVector.x);
+
+  const vx = Math.cos(angle);
+  const vy = Math.sin(angle);
+
+  if (vx === 0) {
+    return x2 - x1;
+  } else if (vy === 0) {
+    return y2 - y1;
+  }
+
+  const leftEdge = (x1 - px) / vx;
+  const rightEdge = (x2 - px) / vx;
+
+  const topEdge = (y1 - py) / vy;
+  const bottomEdge = (y2 - py) / vy;
+
+  return Math.min(...[leftEdge, rightEdge, topEdge, bottomEdge].filter(edge => edge > 0));
+}
+
+/**
+ * Calculate a the vector length from the center of a circle to it's perimeter.
+ *
+ * @param {number} nodeSize The size of the circle, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {boolean} isCustomNode is viewGenerator specified.
+ * @returns {number} The length of the vector from the center of the symbol to it's closet edge, considering the given direction vector.
+ */
+function calcCircleVectorLength(nodeSize, isCustomNode) {
+  let radiusLength;
+  if (isCustomNode) {
+    // nodeSize equals the Diameter in the case of custome-node.
+    radiusLength = nodeSize / 10 / 2;
+  } else {
+    // because this is a circle and A = pi * r^2
+    // we multiply by 0.95, because if we don't the link is not melting properly
+    radiusLength = Math.sqrt(nodeSize / Math.PI);
+  }
+  return radiusLength;
+}
+
+/**
+ * Calculate a the vector length from the center of a square to it's perimeter.
```

Comment:
Calculates the vector length from the center of a square to the closest edge following a direction.